designs/data-lake.md

```diff
@@ -0,0 +1,114 @@
+# Design for FlowEHR Data Lake
+
+| Author                     | Issue                                              |
+|----------------------------|----------------------------------------------------|
+| James Griffin (@jjgriff93) | https://github.com/UCLH-Foundry/FlowEHR/issues/160 |
+
+
+## Context
+
+FlowEHR enables data engineers to develop pipelines that transform data from raw sources into a high-quality data store that powers analytics, machine learning and more. In a medallion structure, we would consider this a transformation from a "bronze" layer through to "gold".
+
+Currently, FlowEHR provides infrastructure for running transformations (via Databricks) and for storing features (the SQL feature store which we can also consider the "gold" layer); however, it does not provide mechanisms for storing the intermediary stages of these transformations from raw to gold, and is not opinionated on how this data is stored.
+
+
+## Proposal
+
+To make it as easy as possible for data engineers to get started in developing pipelines, and to provide an existing storage structure to enforce industry best practices out-of-the-box, I propose the introduction of a datalake component to the FlowEHR platform.
+
+### What is a Data Lake?
+
+A data lake is a massively scalable and secure data storage for high-performance analytics workloads. Due to the volume of data that modern organisations need to deal with, data lakes have become a defacto choice for storing, analysing and consuming data at scale.
```

Comment:
I think somewhere here we should also talk about using Delta tables for storage and the benefits it brings. Happy to contribute this section if you like

Reply:
Looks like my previous comment didn't save - yep feel free to add a quick section, I call it out in the considerations but is a good out of the box benefit we could use when data lake is deployed